gammapy/maps/core.py

```diff
@@ -1088,9 +1088,17 @@ def reproject_to_geom(self, geom, preserve_counts=False, precision_factor=10):
             output_map = output_map.resample(geom, preserve_counts=preserve_counts)
         return output_map
 
-    def fill_events(self, events):
-        """Fill event coordinates (`~gammapy.data.EventList`)."""
-        self.fill_by_coord(events.map_coord(self.geom))
+    def fill_events(self, events, weights=None):
+        """Fill event coordinates (`~gammapy.data.EventList`).
+
+        Parameters
+        ----------
+        events : `~gammapy.data.EventList`
+            Events to be fill in the map.
+        weights : `~numpy.ndarray`
+            Weights vector. Default is weight of one.
```

Comment:
Add that it should have the same shape as `events` or be broadcastable (not sure what is correct...)